src/Symfony/Component/Form/Extension/Core/Type/ChoiceType.php

```diff
@@ -390,6 +404,10 @@ private function addSubForm(FormBuilderInterface $builder, string $name, ChoiceV
     private function createChoiceList(array $options)
     {
         if (null !== $options['choice_loader']) {
+            if (null !== $options['choice_filter'] && $options['choice_loader'] instanceof ChoiceFilterInterface) {
```

Comment:
i think something should happen when the choice loader does not support `ChoiceFilterInterface`. E.g. wrap it in a decorating / callback loader, or throw.

Reply:
yeah, I think throwing an exception is a good DX :+1: